### DISCUSSION TITLE
[Feature Request] Better handling of vault token caching

### DISCUSSION THREAD

**Discussion by dwoz** [saltstack/salt#67376] [Feature Requests]
**Description**

This fix prevents crashes but the token caching mechanism still broken.

https://github.com/saltstack/salt/pull/62195

When multiple threads are trying to access the vault token the 'cache' gets overwrote causing the threads to still request a token from vault. Effectively making the 'cache' useless.
  
**Steps to Reproduce the behavior**
(Include debug logs if possible and relevant)

**Expected behavior**

**Screenshots**

**Versions Report**

All current versions.

**Comment by garethgreenaway** [comment]
Determined this to be a larger issue that originally thought with additional requirements, pushing this out to 3007.
